landside/catkin_ws/src/gui/README.md

```diff
@@ -86,7 +85,125 @@ The table will show you what nodes are running.
+
+To use this capability, add the `doc` attribute to `<arg>` tag in your launch file. Inside this `doc` attribute, put a JSON object.
+
+Example:
+```xml
+<arg name="count" default="" doc='{"type":"int","allowEmpty"=false,"help":"The number of messages to publish"}'>
+```
+**There must NOT be anything other than the JSON object in the `doc` attribute for the validation to work.**
+
+If there is an error in parsing the JSON, the user will be presented with an unrestricted text input.
+
+The following table lists all attributes and values that can be part of the JSON object that are interpreted by this plugin. For each combination of attribute and possible value, the "Impact" column describes the change in the plugin's UI or behavior that will result from the combination being part of the JSON object. This "Impact" will materialize _if and only if_ the argument's default value is either not set or is part of the "Accepted Defaults" for the combination of attribute and possible value. For combinations that have a listed tooltip, the tooltip will be displayed when the user hovers over the argument name.
+
+<table>
+<thead>
+  <tr>
+    <th>Attribute</th>
+    <th>Possible Values</th>
+    <th>Impact</th>
+    <th>Accepted Defaults</th>
+    <th>Tooltip</th>
+  </tr>
+</thead>
+<tbody>
+  <tr>
+    <td rowspan="4">type</td>
+    <td>"int"</td>
+    <td>Text input restricted to integers only</td>
+    <td>Any integer or empty string</td>
+    <td>"Type: int"</td>
+  </tr>
+  <tr>
+    <td>"double"</td>
+    <td>Text input restricted to floating point values only</td>
+    <td>Any floating point value or empty string</td>
+    <td>"Type: double"</td>
+  </tr>
+  <tr>
+    <td>"bool"</td>
+    <td>Checkbox</td>
+    <td>"true" or "false" (case-insensitive)</td>
+    <td>"Type: bool"</td>
+  </tr>
+  <tr>
+    <td>"str"</td>
+    <td>Text input unrestricted</td>
+    <td>Any string</td>
+    <td>"Type: str"</td>
+  </tr>
+  <tr>
+    <td>regex</td>
+    <td>Any valid regex in a string*</td>
+    <td>Text input restricted to match regex</td>
+    <td>Any string matching the regex or empty string</td>
+    <td>"Regex: [provided_regex]"</td>
+  </tr>
+  <tr>
+    <td>options</td>
+    <td>Any non-empty list of strings</td>
+    <td>Dropdown only containing values from list</td>
+    <td>Any value in the list**</td>
+    <td>None</td>
+  </tr>
+  <tr>
+    <td>help</td>
+    <td>Any non-empty string</td>
+    <td>See Tooltip</td>
+    <td>N/A</td>
+    <td>"Help: [provided_help_text]"</td>
+  </tr>
+  <tr>
+    <td rowspan="2">allowEmpty</td>
+    <td>true (no quotes, case-sensitive)</td>
+    <td>Plugin allows argument value to be empty string†</td>
+    <td>Empty string†</td>
+    <td>None</td>
+  </tr>
+  <tr>
+    <td>false (no quotes, case-sensitive)</td>
+    <td>Plugin requires argument value to be a empty string</td>
+    <td>Any string</td>
+    <td>None</td>
+  </tr>
+</tbody>
+</table>
```

Comment:
Might be good to document default values of these attributes. Or at least for `allowEmpty`, since the others are obvious

Reply:
The default value for `allowEmpty` is mentioned in the bullet points under "A few more notes about the table".

For all attributes other than `allowEmpty`, if they are not specified they don't have a default value. The plugin behaves differently if an attribute is specified with any value versus if the attribute is not specified at all. Saying that the attributes have default values would imply that the plugin always behaves as if all the attributes have values, which is not true.